update

LianMiny opened the pull request:
**Features**
<!-- Describe the features or bug fixes in this PR. For bug fixes, link to the issue. -->

- Feature 1
- Feature 2

**Feature Docs**
<!-- Provide RFC, tutorial, or use case links for significant updates. Optional for minor changes. -->

**Influence**
<!-- Explain the impact of these changes for reviewer focus. -->

**Result**
<!-- Include screenshots or logs of unit tests or running results. -->

**Other**
<!-- Additional notes about this PR. -->

Randool committed:
feat: Adding human interaction functionality to Manus

666haiwen committed:
feat: add chart visualization tools which support png/html output

666haiwen committed:
feat: remove useless test code

666haiwen committed:
feat: extract utils function in chart generation tools and optimize tool description

666haiwen committed:
feat: add data analysis agent with chart generation tools

666haiwen committed:
refactor: update code formatter

mannaandpoem committed:
Merge pull request #861 from 666haiwen/feat/chart-visualization

feat: add chart visualization tools which support png/html output

666haiwen committed:
Merge branch 'feat/chart-visualization' into feat/data_visualization_hackathon

666haiwen committed:
feat: add test demo

Commit:
I have implemented a fix to asyncio resource leak errors when the script finishes. A cleanup method has been added to the agent hierarchy (ToolCallAgent, BrowserAgent, Manus), and main.py now ensures this cleanup is called using a try...finally block before the program exits. This should prevent the ValueError: I/O operation on closed pipe exceptions by properly closing browser resources.

Commit:
precommit isort change

zyren123 committed:
Remove redundant code

zyren123 committed:
Fix bugs next_step_prompt accidentally overwritten

666haiwen committed:
feat: add chart prepare tools and update chart geneartion tool to generate chart parallel

ZhangZixunCodeSpace committed:
feat: generate structured analysis reports

- Add data_exploration.md with dataset metadata and statistics
- Create preprocessing_result.md for cleaning logs and metrics
- Ensure markdown-only outputs without visual elements

ZhangZixunCodeSpace committed:
Merge branch 'feat/data_visualization_hackathon' of https://github.com/666haiwen/OpenManus into feat/data_visualization_hackathon
Merge remote-tracking branch 'feat/data_visualization_hackathon'

Sync local work with upstream changes from hackathon branch

666haiwen committed:
Revert "Merge branch 'feat/data_visualization_hackathon' of https://github.com/666haiwen/OpenManus into feat/data_visualization_hackathon"

This reverts commit b0e9384502cad2a19f32298327524619253255ea, reversing
changes made to 96c23f1f56448efab8b2d866c897a49a01a39f60.

666haiwen committed:
Revert "feat: generate structured analysis reports"

This reverts commit 96c23f1f56448efab8b2d866c897a49a01a39f60.

a-holm committed:
Merge branch 'mannaandpoem:main' into fix-error-with-cleaning-up

mannaandpoem committed:
remove unused tools

mannaandpoem committed:
Merge pull request #954 from zyren123/fixbug

Fix manus agent's next step prompt accidentally overwritten by browser agent

mannaandpoem committed:
Merge pull request #951 from a-holm/fix-error-with-cleaning-up

Fix asyncio resource leaks by implementing agent cleanup hierarchy

mannaandpoem committed:
fix bug of type hint

mannaandpoem committed:
update deep_research.py

mannaandpoem committed:
fix bug of stream for ask_tool

mannaandpoem committed:
update extract_content action for BrowserUseTool

mannaandpoem committed:
update DeepResearch, ResearchSummary and WebContentFetcher

mannaandpoem committed:
update config.example-model-*.toml

mannaandpoem committed:
update WebSearch and DeepResearch

mannaandpoem committed:
update Manus, BrowserAgent and ToolCallAgent

mannaandpoem committed:
update web_search action

666haiwen committed:
feat: update data analysis agent prompt and output content

666haiwen committed:
Merge branch 'feat/data_visualization_hackathon' into feat/data_visualization_hack_czx

666haiwen committed:
fix: remove wrong default prompt in function tool

666haiwen committed:
Merge pull request #1 from 666haiwen/feat/data_visualization_hack_czx

Feat: Update chart generation tools in data analysis

ZhangZixunCodeSpace committed:
feat(zzx): add data preprocessing module

666haiwen committed:
Merge branch 'feat/data_visualization_hackathon' into feat/data_visualization_hackathon_zzx

mannaandpoem committed:
update _BROWSER_DESCRIPTION

ZhangZixunCodeSpace committed:
Feat: Update data preprocessing and exploration function in data analysis tool

666haiwen committed:
feat: add specInsight func / language params in data visualization tool

666haiwen committed:
feat: update prompt and hint

didiforgithub committed:
Add Sponsors to ReadMe.

ZhangZixunCodeSpace committed:
Feat: delete unrequired part in data analysis tool

666haiwen committed:
Merge pull request #2 from 666haiwen/feat/data_visualization_hackathon_zzx

Feat/data visualization hackathon zzx

666haiwen committed:
Merge remote-tracking branch 'origin/feat/data_visualization_hackathon' into feat/data_visualization_hack_czx

666haiwen committed:
Merge pull request #4 from 666haiwen/feat/data_visualization_hack_czx

Feat/data visualization hack czx

666haiwen committed:
feat: update prompt of data analysis agent

666haiwen committed:
feat: update vmind version

666haiwen committed:
Merge pull request #5 from 666haiwen/feat/data_visualization_hack_czx

Feat/data visualization hack czx

666haiwen committed:
Merge remote-tracking branch 'upstream/main' into feat/data_visualization_hackathon

666haiwen committed:
feat: update hack_demo

666haiwen committed:
feat: update hack_demo to data analysis agent

666haiwen committed:
fix: fix package.lock bnpm issus

666haiwen committed:
fix: fix bse64 png show error

666haiwen committed:
feat: update prompt and test of hack demo

666haiwen committed:
feat: optimize chart generation file name

666haiwen committed:
feat: update prompt

666haiwen committed:
feat: revert insights hint in next steps

666haiwen committed:
Revert "feat: revert insights hint in next steps"

This reverts commit 9dbc2306c8ac48ffc6bcba6fd65ac8f22272ea74.

666haiwen committed:
Revert "feat: update prompt"

This reverts commit 235b7af89dfc2a2f45d960000af9a4aad992a13e.

666haiwen committed:
Revert "feat: optimize chart generation file name"

This reverts commit f1cad2f69e5dbcebaccc249f565c4cc6ed89b36d.

666haiwen committed:
feat: add default params + transfer vchart.min.js to cdn

666haiwen committed:
feat: cdn update to default

666haiwen committed:
feat: update readme and demo data

mannaandpoem committed:
Add DOI badge

lmmmmmmmmmmmmmmm commented:
> **Features**
> 
> * Feature 1
> * Feature 2
> 
> **Feature Docs**
> 
> **Influence**
> 
> **Result**
> 
> **Other**

Commit:
add terminate for NEXT_STEP_PROMPT in browser & manus

fred913 committed:
Merge pull request #1050 from Avalephh/addPrompt

add terminate for NEXT_STEP_PROMPT in browser & manus

cyzus committed:
1. support multiple mcp servers
2. allow manus agent to connect to mcp servers whilst keeping the local tool set
3. include automatic mcp config loading

cyzus committed:
if mcp.json is not provided, do not report anything

cyzus committed:
make sure tool is not repetitively added because mcp servers may provide the same tool

cyzus committed:
catch mcp cancel scope issue

cyzus committed:
avoid repetitive mcp tool adding

cyzus committed:
fix cancel scope issue

cyzus committed:
remove session close

cnJasonZ committed:
feat: add config example for PPIO

mannaandpoem committed:
Merge pull request #1067 from cnJasonZ/feat/ppio-config-example

feat: add config example for PPIO

mannaandpoem committed:
Merge pull request #1061 from cyzus/feat-include-mcp-in-manus-agent

Manus Agent to include MCP tools

mannaandpoem committed:
Merge pull request #116 from Randool/dev/add_human_interact

feat: Adding human interaction functionality to Manus

fred913 committed:
chore(github): correct config.yml

zhenjunMa committed:
delete duplicate code

666haiwen committed:
feat: resize steps limits

666haiwen committed:
Merge remote-tracking branch 'upstream/main' into feat/data_visualization_hackathon

666haiwen committed:
feat: refactor code to prepare merge into main

666haiwen committed:
feat: update demo and fix spec string bug in chart visualization

666haiwen committed:
refactor: remove useless code

666haiwen committed:
refactor: refactor code by pre-commit

mannaandpoem committed:
Merge pull request #1097 from 666haiwen/feat/chart-visualization

feat: add data analysis agent to do data process / data report / chart generation / ... tasks

mannaandpoem committed:
delete useless tools and prompt

mannaandpoem committed:
Merge pull request #1085 from zhenjunMa/delete_duplicate_code

delete duplicate code

CZH-THU committed:
Support A2A protocol

CZH-THU committed:
Change README

zoe-yyx committed:
remove comments

mannaandpoem committed:
Merge pull request #1104 from zoe-yyx/simplify-code

Simplify return statement in count_image by removing redundant comments

666haiwen committed:
fix: fix wrong resolved url in pacakges-lock.json

CZH-THU committed:
Merge branch 'mannaandpoem:main' into main

CZH-THU committed:
use Manus.create() to create instance

CZH-THU committed:
use factory to create Manus Agent instance dynamically

CZH-THU committed:
add requirements.txt

JackLuguibin committed:
BUGFIX: FIX MCPAgent Bug

mannaandpoem committed:
format code

CZH-THU committed:
Merge branch 'FoundationAgents:main' into main

didiforgithub committed:
Update Project URL.

CZH-THU committed:
Merge branch 'FoundationAgents:main' into main

666haiwen committed:
feat: add data_analysis_agent into run_flow && using puppeteer instead of canvas in package

666haiwen committed:
fix: refactor code by pre-commit

didiforgithub committed:
Merge pull request #1139 from 666haiwen/feat/add_agent_in_runflow

Feat/add data analysis agent in runflow

CZH-THU committed:
Merge branch 'main' of https://github.com/FoundationAgents/OpenManus

CZH-THU committed:
Merge branch 'FoundationAgents:main' into main

CZH-THU committed:
Merge branch 'main' of https://github.com/CZH-THU/OpenManus

CZH-THU committed:
[feat]:Support A2A protocol with a2a-sdk

ryantzr1 committed:
fix: sanitize MCP tool names for OpenAI API compliance

Tool names with invalid characters (slashes, dots) from server paths
were causing OpenAI API 400 errors. Added sanitization to ensure
names match required pattern ^[a-zA-Z0-9_-]{1,64}$.

Fixes #1143

Kunlun-Zhu committed:
add prompt input parser into main.py

Kunlun-Zhu committed:
fixing isort errors

didiforgithub committed:
Merge pull request #1153 from Kunlun-Zhu/main

Add prompt input parser into main.py

didiforgithub committed:
Merge pull request #1145 from ryantzr1/main

fix: sanitize MCP tool names for OpenAI API compliance

CZH-THU committed:
Merge branch 'FoundationAgents:main' into main

CZH-THU committed:
feat:support A2A protocol

CZH-THU committed:
parse host&port with argparse and reformat files

CZH-THU committed:
sandbox init

CZH-THU committed:
add utils

white-rm committed:
add sandox utils

CZH-THU committed:
modify BaseTool

CZH-THU committed:
add sandbox tool

CZH-THU committed:
debug computer use tool

white-rm committed:
add shell sandox

CZH-THU committed:
add daytona config

CZH-THU committed:
Merge branch 'sandbox' of https://github.com/XYDT-AI/OpenManus-sandbox into sandbox

CZH-THU committed:
test sb_browser_use_tool

white-rm committed:
update shell sandox

CZH-THU committed:
init sandbox agent

CZH-THU committed:
Merge branch 'sandbox' of https://github.com/XYDT-AI/OpenManus-sandbox into sandbox

CZH-THU committed:
specify a2a-sdk version

white-rm committed:
update browser sandbox tool

didiforgithub committed:
Merge pull request #1101 from CZH-THU/main

feat:Support A2A protocol

white-rm committed:
update browser sandbox tool message

white-rm committed:
update browser sandbox tool 2

cyb1278588254 committed:
init sb_files_tool & sb_vision_tool

CZH-THU committed:
add main for sandbox_agent

CZH-THU committed:
Merge branch 'FoundationAgents:main' into sandbox

CZH-THU committed:
support use existing sandbox by id

CZH-THU committed:
Merge branch 'sandbox' of https://github.com/XYDT-AI/OpenManus-sandbox into sandbox

cyb1278588254 committed:
debug sb_files_tool & sb_vision_tool

SNHuan committed:
add Crawl4aiTool

white-rm committed:
eliminate sandox redundant information

SNHuan committed:
Format crawl4ai.py with pre-commit hooks

didiforgithub committed:
Merge pull request #1197 from SNHuan/tool/Crawl4aiTool

add Crawl4aiTool

CZH-THU committed:
add README and delete sandbox_id

CZH-THU committed:
feat:support 4 sandbox tools by using daytona

CZH-THU committed:
Merge branch 'FoundationAgents:main' into sandbox

CZH-THU committed:
Apply pre-commit formatting fixes

CZH-THU committed:
Merge branch 'sandbox' of https://github.com/XYDT-AI/OpenManus-sandbox into sandbox

CZH-THU committed:
Apply pre-commit formatting fixes

CZH-THU committed:
remove sb_*.py to tool/sandbox&modify requirements.txt

CZH-THU committed:
modify requirements.txt

didiforgithub committed:
Update Group Link.

CZH-THU committed:
modify sb_browser_use tool's result

CZH-THU committed:
Merge branch 'FoundationAgents:main' into sandbox

didiforgithub committed:
Update README.md

CZH-THU committed:
Merge branch 'FoundationAgents:main' into sandbox

CZH-THU committed:
put base64 in the currect position for sb_vision

CZH-THU committed:
Merge branch 'sandbox' of https://github.com/XYDT-AI/OpenManus-improve into sandbox

didiforgithub committed:
Merge pull request #1206 from XYDT-AI/sandbox

Feat: support 4 sandbox tools by using daytona

didiforgithub committed:
Merge pull request #1256 from FoundationAgents/chart-visualization

Merge pull request #861 from 666haiwen/feat/chart-visualization

cnJasonZ committed:
feat: add JiekouAI as new provider

didiforgithub committed:
Merge pull request #1266 from cnJasonZ/feat/jiekouAI

Add Jiekou.AI as new LLM provider

didiforgithub committed:
Update README.md

MagMueller committed:
feat(browser): use Browser Use CLI 3.0

MagMueller committed:
docs(browser): add CLI setup paths

MagMueller committed:
refactor(browser): use CLI 3.0 MCP server

MagMueller committed:
fix(browser): keep explicit termination

MagMueller committed:
fix(mcp): deliver tool images as user messages

MagMueller committed:
fix(browser): forward CLI connection settings

mannaandpoem committed:
Merge pull request #1398 from MagMueller/agent/browser-use-cli-3

feat(browser): make Browser Use CLI 3.0 MCP the default

Commit:
docs: correct author name

Commit:
fix(deps): add missing structlog and cap pillow below 11
